addons/gamification/views/mail_templates.xml

```diff
@@ -6,6 +6,7 @@
             <field name="model_id" ref="base.model_res_users"/>
             <field name="subject">New rank: ${object.rank_id.name}</field>
             <field name="email_to"></field>
+            <field name="email_from">${user.email_formatted or object.email_formatted}</field>
```

Comment:
Would say
```
${user.email_formatted or object.company_id.partner_id.email_formatted or object.email_formatted | safe}
```
Side note: check that it works in 13 (I looked at templates in master, maybe re-check in 13 to be sure)

Don't forget the | safe in jinja, otherwise you may have encoding issues with formatted emails.